modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/internal/info/collection/provider/test/ManyToManyObjectRelationshipInfoCollectionProviderTest.java

```diff
@@ -0,0 +1,262 @@
+			_relatedObjectDefinition, _objectDefinition,
+			ObjectRelationshipConstants.TYPE_MANY_TO_MANY);
+
+		try {
+			ObjectEntry relatedObjectEntry1 =
+				_objectEntryLocalService.addObjectEntry(
+					TestPropsValues.getUserId(), _group.getGroupId(),
+					_relatedObjectDefinition.getObjectDefinitionId(),
+					HashMapBuilder.<String, Serializable>put(
+						"relatedName", RandomTestUtil.randomString()
+					).build(),
+					ServiceContextTestUtil.getServiceContext());
+
+			ObjectEntry relatedObjectEntry2 =
+				_objectEntryLocalService.addObjectEntry(
+					TestPropsValues.getUserId(), _group.getGroupId(),
+					_relatedObjectDefinition.getObjectDefinitionId(),
+					HashMapBuilder.<String, Serializable>put(
+						"relatedName", RandomTestUtil.randomString()
+					).build(),
+					ServiceContextTestUtil.getServiceContext());
+
+			ObjectEntry objectEntry = _objectEntryLocalService.addObjectEntry(
+				TestPropsValues.getUserId(), _group.getGroupId(),
+				_objectDefinition.getObjectDefinitionId(),
+				HashMapBuilder.<String, Serializable>put(
+					"name", RandomTestUtil.randomString()
+				).build(),
+				ServiceContextTestUtil.getServiceContext());
+
+			_objectRelationshipLocalService.
+				addObjectRelationshipMappingTableValues(
+					TestPropsValues.getUserId(),
+					objectRelationship.getObjectRelationshipId(),
+					objectEntry.getObjectEntryId(),
+					relatedObjectEntry1.getObjectEntryId(),
+					new ServiceContext());
+
+			_objectRelationshipLocalService.
+				addObjectRelationshipMappingTableValues(
+					TestPropsValues.getUserId(),
+					objectRelationship.getObjectRelationshipId(),
+					objectEntry.getObjectEntryId(),
+					relatedObjectEntry2.getObjectEntryId(),
+					new ServiceContext());
+
+			RelatedInfoItemCollectionProvider
+				relatedInfoItemCollectionProvider =
+					_infoItemServiceRegistry.getFirstInfoItemService(
+						RelatedInfoItemCollectionProvider.class,
+						StringBundler.concat(
+							ObjectDefinition.class.getName(), StringPool.POUND,
+							_objectDefinition.getObjectDefinitionId()));
+
+			Assert.assertNotNull(relatedInfoItemCollectionProvider);
+
+			CollectionQuery collectionQuery = new CollectionQuery();
+
+			collectionQuery.setPagination(Pagination.of(2, 0));
+			collectionQuery.setRelatedItemObject(objectEntry);
+
+			InfoPage collectionInfoPage =
+				relatedInfoItemCollectionProvider.getCollectionInfoPage(
+					collectionQuery);
+
+			List<ObjectEntry> objectEntries = collectionInfoPage.getPageItems();
+
+			Assert.assertNotNull(objectEntries);
+
+			Assert.assertEquals(
+				objectEntries.toString(), 2, objectEntries.size());
+
+			Assert.assertEquals(2, collectionInfoPage.getTotalCount());
+
+			Assert.assertTrue(objectEntries.contains(relatedObjectEntry1));
+			Assert.assertTrue(objectEntries.contains(relatedObjectEntry2));
+		}
+		finally {
+			if (objectRelationship != null) {
+				_objectRelationshipLocalService.deleteObjectRelationship(
+					objectRelationship);
```

Comment:
use @DeleteAfterTestRun to delete the `objectRelationship`

Reply:
Hi @MarinhoFeliphe, I tried this, but as I mentioned in my last commit, DataGuard throws an error when deleting the objects